assets/js/components/data.js

```diff
@@ -321,83 +333,132 @@ const data = {
+		// Check variable cache first.
+		if ( 'undefined' !== typeof googlesitekit.admin.datacache[ key ] ) {
+			return googlesitekit.admin.datacache[ key ];
+		}
 
-		// Check the datacache, which only includes fresh data at load.
-		if ( ! cache && datacache && datacache[ 'googlesitekit_' + key ] ) {
-			return datacache[ 'googlesitekit_' + key ];
+		const storage = detectPersistentCache();
+		if ( ! storage ) {
+			return undefined;
 		}
 
-		// Check the datacache, which only includes fresh data at load.
-		if ( ! cache && datacache && datacache[ 'googlesitekit_' + hashlessKey ] ) {
-			return datacache[ 'googlesitekit_' + hashlessKey ];
+		// Check persistent cache.
+		const cache = JSON.parse( window[ storage ].getItem( 'googlesitekit_' + key ) );
+		if ( cache && 'object' === typeof cache && cache.date ) {
+			// Only return value if no maximum age given or value is newer than it.
+			if ( ! maxAge || ( Date.now() / 1000 ) - cache.date < maxAge ) {
+				// Set variable cache.
+				googlesitekit.admin.datacache[ key ] = cache.value;
+
+				return cache.value;
+			}
 		}
 
-		if ( ! cache ) {
-			return false;
+		return undefined;
+	},
+
+	/**
+	 * Removes data from the cache.
+	 *
+	 * @param {string} key The cache key,
+	 */
+	deleteCache( key ) {
+		lazilySetupLocalCache();
+
+		if ( 'undefined' !== typeof googlesitekit.admin.datacache[ key ] ) {
+			delete googlesitekit.admin.datacache[ key ];
 		}
 
-		const age = ( Date.now() / 1000 ) - cache.date;
-		if ( age < maxAge ) {
-			return cache.value;
+		const storage = detectPersistentCache();
+		if ( ! storage ) {
+			return;
 		}
-		return false;
+
+		window[ storage ].removeItem( 'googlesitekit_' + key );
 	},
 
 	/**
-	 * Remove cache.
+	 * Invalidates all caches associated with a specific cache group.
 	 *
-	 * @param {Object} dataObject The data object.
-	 * @param {string} key        The cache key,
+	 * @param {string} type       The data to access. One of 'core' or 'modules'.
+	 * @param {string} identifier The data identifier, for example a module slug.
+	 * @param {string} datapoint  The datapoint.
 	 */
-	deleteCache( dataObject, key ) {
-		if ( window.sessionStorage ) {
-			window.sessionStorage.removeItem( dataObject + '::' + key );
+	invalidateCacheGroup( type, identifier, datapoint ) {
+		const groupPrefix = this.getCacheKey( type, identifier, datapoint );
+
+		lazilySetupLocalCache();
+
+		Object.keys( googlesitekit.admin.datacache ).forEach( ( key ) => {
+			if ( 0 === key.indexOf( groupPrefix + '::' ) || key === groupPrefix ) {
+				delete googlesitekit.admin.datacache[ key ];
+			}
+		} );
+
+		const storage = detectPersistentCache();
+		if ( ! storage ) {
+			return;
 		}
+
+		Object.keys( window[ storage ] ).forEach( ( key ) => {
+			if ( 0 === key.indexOf( 'googlesitekit_' + groupPrefix + '::' ) || key === 'googlesitekit_' + groupPrefix ) {
+				window[ storage ].removeItem( key );
+			}
+		} );
```

Comment:
This pattern of detecting the storage, and using `window[ storage ]` is repeated in many places.
Why not extract this to a utility which gives you an object that does this for you?
e.g.

```js
import { storage } from 'utils';
storage.addItem( ... );
storage.removeItem( ... );
```

If neither local or session storage was available, it could use a no-op. This would simplify the consuming code quite a bit.

Reply:
Sounds great! I just added this change, with the minor difference that I still keep logic to check whether a storage is available. This allows to not unnecessarily run other logic that won't actually be used.